### PR TITLE
Added Support for keys with array values

### LINKED ISSUE #11: Support for multiple values
Include support for multiple value section. 

eg:- what env do you support 
1. Browser
2. Node
3. Jasmine 
4.  jquery

Here i could select more than one. Also, allow me to decide if the key is array or object.

### PR DESCRIPTION
**Pull request by awebdeveloper**
Now array values are supported in which case value will get pushed to the array. This(push) can be overridden to replace in either of the two case.

1. By setting replace: true on a step
2. By setting replace: true on a step variant 

partly fixes #11

### CODE REVIEW

File: oncogene.js [hunk 1]
```diff
@@ -54,7 +54,7 @@ class Oncogene {
         }
     }
 
-    setVal(key, value) {
+    setVal(key, value, selectedVarient, step) {
```
Comment (gwer) [inline oncogene.js]: You can use `this.getStep()` instead of passing step argument.

**Review by gwer — CHANGES_REQUESTED** [review]
I think the replacement should be default behaviour. 

What if we will not using `replace` flag? What about something like boolean `push` inside variants[]? Not in steps[].

If the old value is an array, the selected value will be pushed into it. If the old value is not exists, the selected value will be wrapped into an array and added to the config. If the old value is not an array, it will be replaced with the selected value wrapped into an array. If the selected variant has no `push` flag, it always will replace the old value. 

What do you think about it?

**Comment by awebdeveloper** [comment]
Makes sense. I will instead add push: true.

Push can be a value of both varietal and step too so as to allow a variety of cases. I could explain further if you have doubts


Will look into getStep()

**Comment by awebdeveloper** [comment]
Changes done

**Comment by gwer** [comment]
If `options.steps[].push === true` and `options.steps[].variants[].push === false` then new value will be pushed into array? But `options.steps[].variants[].push` has greater specificity...

Let's leave only one thing. I vote for `options.steps[].variants[].push` because it's more flexible. I think adding a flag to each variant is not a big problem.

**Comment by awebdeveloper** [comment]
k done

**Review by gwer — APPROVED** [review]

**Comment by gwer** [comment]
Looks fine. Thanks.

**Pull request merged**